pkg/controllers/clusterresourceplacement/clusterresourceplacement_controller.go

```diff
@@ -17,27 +17,51 @@ import (
 	"go.goms.io/fleet/pkg/utils/controller"
 )
 
-func (r *Reconciler) Reconcile(_ context.Context, _ controller.QueueKey) (ctrl.Result, error) {
-	return ctrl.Result{}, nil
+func (r *Reconciler) Reconcile(ctx context.Context, _ controller.QueueKey) (ctrl.Result, error) {
+	// TODO workaround to bypass lint check
+	return r.handleUpdate(ctx, nil)
 }
 
 // handleUpdate handles the create/update clusterResourcePlacement event.
 // It creates corresponding clusterPolicySnapshot and clusterResourceSnapshot if needed and updates the status based on
 // clusterPolicySnapshot status and work status.
 // If the error type is ErrUnexpectedBehavior, the controller will skip the reconciling.
 func (r *Reconciler) handleUpdate(ctx context.Context, crp *fleetv1beta1.ClusterResourcePlacement) (ctrl.Result, error) {
```

Comment:
If CRP has snapshots, shouldn't it have finalizers?

Reply:
will add it in the future PR.